modules/udf/README.md

```diff
@@ -16,11 +18,117 @@ module "dataset" {
   location                    = "US"
 }
 
+local {
+  parse_url_udf_ddl = <<EOT
+  CREATE FUNCTION IF NOT EXISTS parse_url(url STRING, part STRING)
+  AS (
+    CASE
+      -- Return HOST part of the URL.
+      WHEN part = 'HOST' THEN SPLIT(\`${PROJECT_ID}\`.${dataset_name}.check_protocol(url), '/')[OFFSET(2)]
+      WHEN part = 'REF' THEN
+        IF(REGEXP_CONTAINS(url, '#'), SPLIT(\`${PROJECT_ID}\`.${dataset_name}.check_protocol
+        (url), '#')[OFFSET(1)], NULL)
+      WHEN part = 'PROTOCOL' THEN RTRIM(REGEXP_EXTRACT(url, '^[a-zA-Z]+://'), '://')
+      ELSE ''
+    END
+  );
+  EOT
+  udf_from_file = file(${path.module}/path/to/udf_ddl.sql)
+}
+
 module "add_udfs" {
   source = "terraform-google-modules/bigquery/google//modules/udf"
   version = "~> 4.0"
 
   dataset_id = module.dataset.bigquery_dataset.dataset_id
   project_id = module.dataset.bigquery_dataset.project
+  udf_ddl_query = [local.parse_url_udf_ddl, local.udf_from_file, ]
```

Comment:
I noticed a few places seem to be using `udl_ddl_query`. However, the module uses `udl_ddl_queries`. Please make consistent